controller/RestTest.php

```diff
@@ -15,78 +15,124 @@
-            throw new \common_exception_NotImplemented('Only post method is accepted to compile test');
+            throw new NotImplementedException('Only post method is accepted to compile test');
         }
 
         if (!$this->hasRequestParameter(self::REST_IMPORTER_ID)) {
-            throw new \common_exception_MissingParameter(self::REST_IMPORTER_ID, $this->getRequestURI());
+            throw new MissingParameterException(self::REST_IMPORTER_ID, $this->getRequestURI());
         }
 
-        if (\tao_helpers_Http::hasUploadedFile(self::REST_FILE_NAME)) {
-            $file = \tao_helpers_Http::getUploadedFile(self::REST_FILE_NAME);
-            $importerId = $this->getRequestParameter(self::REST_IMPORTER_ID);
+        if (HttpHelper::hasUploadedFile(self::REST_FILE_NAME)) {
             try {
-                $customParams = [];
-                if ($this->hasRequestParameter(self::REST_DELIVERY_PARAMS)) {
-                    $customParams = $this->getRequestParameter(self::REST_DELIVERY_PARAMS);
-                    $customParams = json_decode(html_entity_decode($customParams), true);
-                }
-                $deliveryClassLabel = '';
-                if ($this->hasRequestParameter(self::REST_DELIVERY_CLASS_LABEL)) {
-                    $deliveryClassLabel = $this->getRequestParameter(self::REST_DELIVERY_CLASS_LABEL);
-                }
-                $task = ImportAndCompile::createTask($importerId, $file, $customParams, $deliveryClassLabel);
+                $importerId = $this->getParameter(self::REST_IMPORTER_ID);
+                $file = HttpHelper::getUploadedFile(self::REST_FILE_NAME);
+                $customParams = $this->getDecodedParameter(self::REST_DELIVERY_PARAMS);
+                $deliveryClassLabels = $this->getDecodedParameter(self::REST_DELIVERY_CLASS_LABELS);
+
+                $task = ImportAndCompile::createTask($importerId, $file, $customParams, $deliveryClassLabels);
             } catch (ImporterNotFound $e) {
-                $this->returnFailure(new \common_exception_NotFound($e->getMessage()));
+                $this->returnFailure(new NotFoundException($e->getMessage()));
             }
-            
+
             $result = ['reference_id' => $task->getId()];
 
             /** @var TaskLogInterface $taskLog */
             $taskLog = $this->getServiceLocator()->get(TaskLogInterface::SERVICE_ID);
-
             $report = $taskLog->getReport($task->getId());
 
-            if (!empty($report)) { //already executed
-                if ($report instanceof \common_report_Report) {
-                    //serialize report to array
-                    $report = json_encode($report);
-                    $report = json_decode($report);
+            if (!empty($report)) {
+                if ($report instanceof Report) {
+                    // Serialize report to array
+                    $report = json_decode(json_encode($report));
```

Comment:
In `json_decode` missed the second parameter which converts an object to an array, by default it converts it so `stdObj`.

Reply:
Done